### PR TITLE
sys-auth/sssd: Add 2.10 

### PR DESCRIPTION
**Pull request by salahcoronya**
There are many changes from 2.9 both by upstream and by me

The main change is now sssd runs as the "sssd" user, instead of root. While sssd supported non-root mode before, it wasn't entirely usable, but now in 2.10 it is. The above also means the existing files have to be migrated: After upgrade, sssd must be stopped the following command issued:
```
systemctl stop sssd OR rc-service sssd stop
chown -R sssd:sssd /var/lib/sss
chown -R sssd:sssd /var/log/sssd
chown -R root:sssd etc/sssd # systemd users can skip this one
systemctl daemon-reload # systemd users only
systemctl start sssd OR rc-service sssd start
```
sssd can then be restarted. The ebuild will check if sssd is running and the permissions on the directories, and notify the emerger the commands to issue post-install. 

Upstream change some of the default, in particular, your have a Kerberos LDAP server with a certificate, either you must configure a certificate or set "ldap_id_use_start_tls = false" in your [domain\WHATEVER] section. 

The "man" USE flag is merged. Man pages are now built unconditionally. use {PKG_,}INSTALL_MASK to not install them
The "sudo" flag is merged. "Sudo" support is now unconditional as it adds no dependencies. This requires a slight change to the app-admin/sudo ebuild (included in PR)
The "acl" flag is merged. Non-root support uses file capabilities, since its needed a build time anyway, no point in excluding it from run time
The "subid" flag is merged. sys-apps/shadow is part of the system set, and all versions of it in the tree support it.

Upstream now uses a tmpfiles.d snippet for for the pid file, however they don't install it and "dotmpfiles" copies the "wrong" one, so I wired up the autotools machinery to install it normally. (The same holds true for some Polkit and udev stuff too, but we're not using that yet).

I tested this both on an openrc machine and a systemd machine, and both work.

Comments welcome.
---

Please check all the boxes that apply:

- [X] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [X] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [X] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [X] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: sys-auth/sssd/sssd-2.10.0_beta1.ebuild [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+	python-single-r1 pam systemd tmpfiles toolchain-funcs
+
+MY_P=${P/_beta/-beta}
+MY_PV=${PV/_beta/-beta}
+
+DESCRIPTION="System Security Services Daemon provides access to identity and authentication"
+HOMEPAGE="https://github.com/SSSD/sssd"
+if [[ ${PV} != 9999 ]]; then
+	SRC_URI="https://github.com/SSSD/sssd/releases/download/${MY_PV}/${MY_P}.tar.gz"
+#	KEYWORDS="~amd64 ~arm ~arm64 ~hppa ~m68k ~mips ~ppc ~ppc64 ~riscv ~sparc ~x86"
```
Comment (thesamesam) [inline sys-auth/sssd/sssd-2.10.0_beta1.ebuild:21]: Please add a check for *_beta, so:
```
if [[ ${PV} != *_beta* ]] ; then
    KEYWORDS=....
fi
```

File: sys-auth/sssd/sssd-2.10.0_beta1.ebuild [hunk 2]
```diff
@@ -18,7 +18,9 @@ DESCRIPTION="System Security Services Daemon provides access to identity and aut
 HOMEPAGE="https://github.com/SSSD/sssd"
 if [[ ${PV} != 9999 ]]; then
 	SRC_URI="https://github.com/SSSD/sssd/releases/download/${MY_PV}/${MY_P}.tar.gz"
-#	KEYWORDS="~amd64 ~arm ~arm64 ~hppa ~m68k ~mips ~ppc ~ppc64 ~riscv ~sparc ~x86"
+	if [[ ${PV} != *_beta ]]; then
```
Comment (thesamesam) [inline sys-auth/sssd/sssd-2.10.0_beta1.ebuild:21]: `*_beta*`, sorry

Comment (salahcoronya) [reply sys-auth/sssd/sssd-2.10.0_beta1.ebuild]: Fixed now

**Comment by salahcoronya** [comment]
Upstream never intended for the Python dependency to be optional, just the bindings: https://github.com/SSSD/sssd/pull/7443 . However, the build system cannot handle it. Since we're dependant on Python anyway, might as well install the binding to work around the problem. If this is undesirable, it can be fixed, but its not trivial to do so.

**Comment by thesamesam** [comment]
> Upstream never intended for the Python dependency to be optional, just the bindings: [SSSD/sssd#7443](https://github.com/SSSD/sssd/pull/7443) . However, the build system cannot handle it. Since we're dependant on Python anyway, might as well install the binding to work around the problem. If this is undesirable, it can be fixed, but its not trivial to do so.

Sorry, I missed this. I would just install them as well then, just of course make sure to handle it properly w/ Python eclasses. Also note this in the commit message and perhaps a comment in the ebuild.

**Comment by alexey-tikhonov** [comment]
Hi.

Take a note that sssd-2.10.0 "GA" was issued upstream meanwhile:
https://github.com/SSSD/sssd/releases/tag/2.10.0

**Comment by thesamesam** [comment]
I generally don't pay attention to things w/ failing CI unless someone asks for help. Are you stuck on it, or?

**Comment by salahcoronya** [comment]
I can't (or don't know how) fix the CI problem. I dropped the "sudo" flag from sssd because it didn't change any dependencies. sudo support is included by default, but it does not depend on sudo. However, sudo current depends on sys-login/sssd[sudo]. While I did include a change  sys-login/sssd[sudo(+)]. for sudo, it needs a revbump. Because I can't stabilize or delete the current stable sudo in this PR, it complains.

**Comment by thesamesam** [comment]
Please git mv the old ebuilds to change it to (+)

**Comment by thesamesam** [comment]
Squash them as well (the sudo ones) or I can in 5m, thanks

**Comment by thesamesam** [comment]
You're a star, thanks

**Comment by thesamesam** [comment]
Mind doing a quick PR to add that UID/GID to api.git too please? (github.com/gentoo/api-gentoo-org)

**Pull request closed without merging**